consensus/proto_array/src/proto_array.rs

```diff
@@ -286,11 +286,15 @@ impl ProtoArray {
                     // Invalid nodes (or their ancestors) should not receive a proposer boost.
                     && !execution_status_is_invalid
                 {
-                    proposer_score = calculate_committee_fraction::<E>(
-                        new_justified_balances,
-                        proposer_score_boost,
-                    )
-                    .ok_or(Error::ProposerBoostOverflow(node_index))?;
+                    // only compute `proposer_score` once
```

Comment:
This already only happens once because of the `proposer_boost_root == node.root` check above. There is only one node in the graph that is eligible for the proposer boost at any given time.

Reply:
Thanks for telling, I reverted this change.

Reply:
There's another small fix, more about coding style, only update when it's changed. This makes align with [ForkChoice.update_checkpoint](https://github.com/sigp/lighthouse/blob/dfcb3363c757671eb19d5f8e519b4b94ac74677a/consensus/fork_choice/src/fork_choice.rs#L946-L956).

Reply:
I don't think that change is really worth making seeing as it's more LoC, and `apply_score_changes` already reads substantially differently from the way the spec does. I'm kind of thinking "if it ain't broke, don't fix it".